packages/contracts/src/contracts/current/protocol/AssetProxyDispatcher/proxies/ERC721Proxy.sol

```diff
@@ -0,0 +1,93 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+
+import "../IAssetProxy.sol";
+import "../../../utils/LibBytes/LibBytes.sol";
+import "../../../utils/Authorizable/Authorizable.sol";
+import "/zeppelin/contracts/token/ERC721/ERC721Token.sol";
+
+contract ERC721Proxy is
+    LibBytes,
+    Authorizable,
+    IAssetProxy
+{
+
+    /// @dev Transfers ERC721 tokens. Either succeeds or throws.
+    /// @param assetMetadata ERC721-encoded byte array
+    /// @param from Address to transfer token from.
+    /// @param to Address to transfer token to.
+    /// @param amount Amount of token to transfer.
+    function transferFrom(
+        bytes assetMetadata,
+        address from,
+        address to,
+        uint256 amount)
+        external
+        onlyAuthorized
+    {
+        // There exists only 1 of each token.
+        require(amount == 1);
+
+        // Decode metadata
+        address token;
+        uint256 tokenId;
+        (token, tokenId) = decodeMetadata(assetMetadata);
+
+        // Either succeeds or throws.
+        ERC721Token(token).transferFrom(from, to, tokenId);
```

Comment:
Should we call safeTransferFrom here? At this current moment we assume `taker` is a user controlled address, but I expect to see contracts integrating with 0x (forwarding contract) and argue it's best to go with the safe way to transfer, lest ERC721 tokens get stuck! 

Thoughts?